lib/sqlalchemy/sql/sqltypes.py

```diff
@@ -3268,6 +3269,97 @@ class MatchType(Boolean):
     """
 
 
+class UUID(TypeEngine):
+
+    """Represent a SQL UUID type.
+
+    Represents the UUID column type, interpreting
+    data either as natively returned by the DBAPI
+    or as Python uuid objects.
+
+    .. note::  :class:`_types.UUID`
+       is provided as a facade for vendor-specific
+       UUID types.  It only works on backends that have an actual
+       UUID type, currently:
+
+       * PostgreSQL
+
+       * MySQL
+
+       * MariaDB
+
+    The UUID type is currently known to work within the prominent DBAPI
+    drivers supported by SQLAlchemy including psycopg, psycopg2, pg8000 and
+    asyncpg. Support for other DBAPI drivers may be incomplete or non-present.
+
+    """
+
+    __visit_name__ = "UUID"
+
+    def __init__(self, as_uuid=True):
+        """Construct a UUID type.
+
+
+        :param as_uuid=True: if True, values will be interpreted
+         as Python uuid objects, converting to/from string via the
+         DBAPI.
+
+         .. versionchanged: 2 ``as_uuid`` now defaults to ``True``.
+
+        """
+        self.as_uuid = as_uuid
+
+    def coerce_compared_value(self, op, value):
+        """See :meth:`.TypeEngine.coerce_compared_value` for a description."""
+
+        if isinstance(value, str):
+            return self
+        else:
+            return super(UUID, self).coerce_compared_value(op, value)
+
+    def bind_processor(self, dialect):
+        if self.as_uuid:
+
+            def process(value):
+                if value is not None:
+                    value = str(value)
+                return value
+
+            return process
+        else:
+            return None
+
+    def result_processor(self, dialect, coltype):
+        if self.as_uuid:
+
+            def process(value):
+                if value is not None:
+                    value = _python_UUID(value)
+                return value
+
+            return process
+        else:
+            return None
+
+    def literal_processor(self, dialect):
+        if self.as_uuid:
+
+            def process(value):
+                if value is not None:
+                    value = "'%s'::UUID" % value
```

Comment:
I don't think we can use this hare, since this is a postgresql specific syntax

Reply:
Thanks Federico, this is related to what we briefly discussed on gitter

![image](https://user-images.githubusercontent.com/13724985/163875973-9f3f500d-41a8-409c-8f34-1507aa5430fb.png)

I guess I'm not completely clear - the purpose of `literal_processor` is to convert a python datatype to the format it will be inserted/selected as in DML? If so, I guess mariaDB doesn't need a special one, it just uses the string representation of a UUID.

Reply:
In that reply I meant that type of `select`, not the `::uuid` cast.

> the purpose of literal_processor is to convert a python datatype to the format it will be inserted/selected as in DML

it's run when a query that specified listeral_bind=True is run. From my limited experience with mariadb/mysql I too don't think there is anything particular to do for it, it's usually PG that's particularly picky about types.

I think we could have here only the version that stringifies the uuid, while the pg specific implementation could override this method to have the cast

Reply:
That's easy enough, I'll change that